packages/web-forms/src/components/controls/Input/InputDate.vue

```diff
@@ -0,0 +1,49 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+import DatePicker from 'primevue/datepicker';
+import type { DateInputNode } from '@getodk/xforms-engine';
+import { ISO_DATE_LIKE_PATTERN } from '@getodk/common/constants/datetime.ts';
+
+interface InputDateProps {
+	readonly question: DateInputNode;
+}
+
+const props = defineProps<InputDateProps>();
+
+const value = computed({
+	get: () => {
+		if (props.question.currentState.value == null) {
+			return null;
+		}
+
+		const temporalValue = props.question.currentState.value.toString();
+		if (!ISO_DATE_LIKE_PATTERN.test(temporalValue)) {
+			return null;
+		}
```

Comment:
Given that we know the question is a `DateInputNode`, can't we use `value` directly and know that will be a `DatetimeRuntimeValue`? Or use `decodeValue` to get a string rather than `toString`?

Maybe the functionality to convert to Date should also be in the engine? Less sure about this one because it really is just for the UI.

Reply:
Using a string (export type DatetimeRuntimeValue = string | null;) is fine for the current needs of the Web Form client, since we don't expect to perform any calculations where the precision of T`emporal.PlainDate` would have been nice.

The original intention was to expose the Temporal classes and let the front-end handle things as needed, providing more flexibility.

What do you think?

Reply:
Given that Temporal is not natively available in Browsers yet. Now I'm more inclined not to expose any Temporal class to the clients, to reduce incompatibilities and put us in a weird spot of support. 

In theory, we control these things more since we manage central releases.

Reply:
I see the benefits of having the client work with strings for now. To get that string, can we use the codec's `decodeValue`?

Reply:
Yes, we can. I'm thinking of returning `YYYY-MM-DDTHH:MM:SS` even when it's a "Date" only (the time is: `T00:00:00`), which will reduce further checks in the front-end.